src/query/functions/utils/heap.go

```diff
@@ -36,14 +38,29 @@ func maxHeapLess(i, j ValueIndexPair) bool {
 	if i.Val == j.Val {
 		return i.Index > j.Index
 	}
-	return i.Val < j.Val
+	return math.IsNaN(i.Val) && !math.IsNaN(j.Val) || i.Val < j.Val
 }
 
 func minHeapLess(i, j ValueIndexPair) bool {
 	if i.Val == j.Val {
 		return i.Index > j.Index
 	}
-	return i.Val > j.Val
+	return i.Val > j.Val || math.IsNaN(i.Val) && !math.IsNaN(j.Val)
+}
+
+func AscFloat64(i, j float64) bool {
+	return i < j || math.IsNaN(j) && !math.IsNaN(i)
+}
+
+func DescFloat64(i, j float64) bool {
+	return i > j || math.IsNaN(j) && !math.IsNaN(i)
+}
+
+func Min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
```

Comment:
nit: This doesn't really need to be a function

Reply:
What could you suggest instead?
To me, the code is more readable this way.